doc/company.texi

```diff
@@ -0,0 +1,1499 @@
+following example:
+
+@lisp
+(setq company-idle-delay 0
+      company-tooltip-idle-delay 10
+      company-require-match nil
+      company-frontends
+      '(company-pseudo-tooltip-unless-just-one-frontend-with-delay
+        company-preview-frontend
+        company-echo-metadata-frontend)
+      company-backends '(company-capf))
+
+(global-set-key (kbd "<tab>")
+                (lambda ()
+                  (interactive)
+                  (let ((company-tooltip-idle-delay 0.0))
+                    (company-complete)
+                    (and company-candidates
+                         (company-call-frontends 'post-command)))))
+@end lisp
+@end defun
+
+@subheading User Options
+
+@macro img{name}
+@ifnothtml
+@center @image{./images/small/\name\,,,,.png}
+@end ifnothtml
+@ifhtml
+@image{./images/large/\name\,,,,.png}
+@end ifhtml
+@end macro
+
+@cindex custom
+@cindex configure
+@cindex interface
+To change the @emph{tooltip frontends} configuration, adjust the
+following user options.
+
+@defopt company-tooltip-align-annotations
+@cindex annotation
+An @dfn{annotation} is a string that carries additional information
+about a candidate; such as a data type, function arguments, or
+whatever a backend appoints to be a valuable piece of information
+about a candidate.  By default, the annotations are shown right beside
+the candidates.  Setting the option value to @code{t} aligns
+annotations to the right side of the tooltip.
+
+@lisp
+(setq company-tooltip-align-annotations t)
+@end lisp
+
+@img{tooltip-annotations}
+
+@end defopt
+
+@defopt company-tooltip-limit
+Controls the maximum number of the candidates shown simultaneously in
+the tooltip (the default value is @code{10}).  When the number of the
+available candidates is larger than this option's value, Company
+paginates the results.
+
+@lisp
+(setq company-tooltip-limit 4)
+@end lisp
+
+@img{tooltip-limit}
+
+@end defopt
+
+@defopt company-tooltip-offset-display
+Use this option to choose in which way to output paginated results.
+The default value is @samp{scrollbar}.  Another supported value is
+@samp{lines}; choose it to show the quantity of the candidates not
+displayed by the current tooltip page.
+
+@lisp
+(setq company-tooltip-offset-display 'lines)
+@end lisp
+
+@img{tooltip-offset-display}
+@end defopt
+
+@c TODO: what is the correct description / behavior?
+@c Maybe improve the docstring.
+@c Does it influence minimal height of the tooltip
+@c together with company-tooltip-limit?
+@c Can we say it ``Controls the position of a tooltip above
+@c or below the point''?
+@c Does this option control two different things: height and position?
+@c @defopt company-tooltip-minimum
+@c Sets the minimum height of a tooltip (also influenced by
+@c @code{company-tooltip-limit}).  If this quantity of lines is not
+@c available in the window, then the tooltip is displayed above the
+@c point.
+
+@c @c This doesn't work the way I expect it to.
```

Comment:
What's the problem?

Reply:
It seems I didn't "re-init" company tooltip (remove letters fully, type again) after scrolling the page and hence was getting wrong positions of the tooltip (above/below the point).